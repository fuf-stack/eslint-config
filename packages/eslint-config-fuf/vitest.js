// TODO: migrate to v0.5.0 + (eslint 9)
// see: https://github.com/veritem/eslint-plugin-vitest

module.exports = {
  plugins: ['vitest'],
  overrides: [
    // test file rules
    {
      files: ['*.spec.ts', '*.spec.tsx', '*.test.ts', '*.test.tsx'],
      extends: ['plugin:vitest/recommended'],
      rules: {
        // Do not enforce production dependencies in tests
        'import/no-extraneous-dependencies': 'off',
      },
    },
  ],
};
