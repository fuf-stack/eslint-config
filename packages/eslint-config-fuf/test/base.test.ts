import { expect, it } from 'vitest';

import { lintFixture } from './helper';

it('fix should add missing semicolons', async () => {
  const { fixedContent } = await lintFixture('base-missing-semicolons.ts');
  expect(fixedContent).toMatchSnapshot();
});

it('unused variables with leading underscore should be allowed', async () => {
  const { fixedContent } = await lintFixture(
    'base-unused-variables-with-leading-underscore.ts',
  );
  expect(fixedContent).toMatchSnapshot();
});
