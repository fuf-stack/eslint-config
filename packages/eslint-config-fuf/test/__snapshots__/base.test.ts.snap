// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fix should add missing semicolons 1`] = `
"// semicolons should be added
const test = 1;
console.log(test);
"
`;

exports[`unused variables with leading underscore should be allowed 1`] = `
"// unused variables with underscores are allowed
const _unused = 'I am unused';

const _alsoUnused = 'Me too';
"
`;
